debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -68,6 +71,9 @@
     private PgConnection typeResolverConnection = null;
     private Long lastCompletelyProcessedLsn;
 
+    private final AtomicLong lastCommittedLsn = new AtomicLong(-1);
```

Comment:
General question: can we make it a habit to comment fields more thoroughly? I'm having a hard time to come up with a 100% clear rule, but I think we should have this for the "important" ones.

E.g. here, does "last committed" refer to committing offsets in KC or in the DB ? And why is it even implemented that way instead of just flushing the LSN from the other thread (in this case I know the answer, just trying to highlight the general kind of questions there might be)? E.g. here we could say this:

```suggestion
    /**
      * The LSN last flushed and applied to the Postgres server. Updated when offsets are committed by 
      * Apache Kafka, i.e. after messages have been sent to the broker and acknowledged. Read and 
      * propagated during the event polling loop, as flushing and polling must happen on the same thread.
    /
    private final AtomicLong lastCommittedLsn = new AtomicLong(-1);
```

Btw. the suggested comment hints that the name isn't quite right; `lastFlushedLsn`perhaps?

Personally, I tend to comment fields and methods and classes, as they are the building blocks one will refer to again and again. These comments also may highlight the higher-level abstractions and design considerations and rationale. I try to be sparse with comments *within* methods, as the need for it often is an indicator for refactoring-worthy implementation (e.g. extraction of sub-methods).

Reply:
@gunnarmorling This variable is set asynchrnounously from `commit` method and when set it is flushed. Hence I believe `lastCommittedLsn` is more expressive. The variable gets cleared right after its value is flushed to replication slot.

Reply:
Ah yes. So my suggested comment isn't quite right in fact. In a nutshell it's exactly these subtle semantics which I think we ought to capture in a better way.

Reply:
@gunnarmorling Comment provided